catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +252,106 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
         return metatypes;
     }
 
+    @Override
+    public List<Serializable> updatePublications(String source, List<String> destinations)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        List<Serializable> updatedPublishedLocations = new ArrayList<>();
+
+        Filter filter = filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .is()
+                .equalTo()
+                .text(source);
+        Query query = new QueryImpl(filter);
+
+        QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(query));
+
+        List<Result> metacards = queryResponse.getResults();
+        if (metacards != null && metacards.size() > 0) {
+            Metacard metacard = metacards.get(0)
+                    .getMetacard();
+            if (metacard != null) {
+                List<Serializable> currentlyPublishedLocations = metacard.getAttribute(
+                        RegistryObjectMetacardType.PUBLISHED_LOCATIONS)
+                        .getValues();
+
+                List<String> publishLocations = destinations;
+                List<String> unpublishLocations = new ArrayList<>();
+
+                unpublishLocations.addAll(currentlyPublishedLocations.stream()
```

Comment:
You should be able to just do:
```
unpublishLocations = currentlyPublishedLocations.stream()...;
```
Instead of adding them all to the array list. That will avoid another loop through to add them again.